packages/store/src/external/hydrate.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+
+import '../types';
+
+import { makeWeakStore } from '../weak-store';
+import { makeStore } from '../store';
+
+/**
+ * @callback MakeBackingStore
+ * @param {HydrateHook} hydrateHook
+ * @returns {BackingStore}
+ */
+
+/**
+ * This creates an external store maker for a given storage backend, supporting
+ * the Closure interface that the rewriter targets.
+ *
+ * @template {Array<any>} A
+ * @template {ExternalInstance} T
+ * @param {MakeBackingStore} makeBackingStore
+ * @returns {MakeHydrateExternalStore<A, T>}
+ */
+export const makeHydrateExternalStoreMaker = makeBackingStore => {
+  const serialize = JSON.stringify;
```

Comment:
If this is intended to demonstrate how one of these would work, fine. But it looks to me like this is intended to be production code. In which case, do *not* use `JSON.stringify`/`JSON.parse` directly. For one thing, it has too many irregularities even on primitive data types: It won't encode `NaN`, `Infinity`, `-Infinity`, `undefined`, or BigInts.

You can use marshal as a fancy data-only serializer by simply insisting that the marshal does not produce any slots, for example by providing a `valueToSlot` function that throws. We might extend from here to a useful valueToSlot in the future. But `JSON.serialize` doesn't handle those cases either, so a data-only marshal would still be superior in all ways to using raw `JSON.stringify`. 

@FUDCo , please double check what I said about JSON above. Thanks.

Reply:
What @erights says about JSON is right.

Reply:
Okay, I will add a dependency on `@agoric/marshal`.

Reply:
> Okay, I will add a dependency on `@agoric/marshal`.

Oh, with my note below, I will push the serialization down into the backing store rather than have it at this layer.  That will enable true flexibility.

Reply:
I like that plan.